src/transformers/pipelines/text_classification.py

```diff
@@ -64,11 +92,38 @@ def __call__(self, *args, **kwargs):
         """
         outputs = super().__call__(*args, **kwargs)
 
-        if self.model.config.num_labels == 1:
-            scores = 1.0 / (1.0 + np.exp(-outputs))
+        return_all_scores = return_all_scores if return_all_scores is not None else self.return_all_scores
+        function_to_apply = function_to_apply if function_to_apply is not None else self.function_to_apply
+
+        if function_to_apply is None and self.model.config.problem_type is not None:
+            if self.model.config.problem_type == "multi_label_classification":
+                self.function_to_apply = "sigmoid"
+            elif self.model.config.problem_type == "single_label_classification":
+                self.function_to_apply = "softmax"
+
+        def sigmoid(_outputs):
+            return 1.0 / (1.0 + np.exp(-_outputs))
+
+        def softmax(_outputs):
+            maxes = np.max(_outputs, axis=-1, keepdims=True)
+            shifted_exp = np.exp(_outputs - maxes)
+            return shifted_exp / shifted_exp.sum(axis=-1, keepdims=True)
+
+        if function_to_apply == "default":
+            if self.model.config.num_labels == 1:
+                scores = sigmoid(outputs)
+            else:
+                scores = softmax(outputs)
+        elif function_to_apply == "sigmoid":
```

Comment:
NIT: we could use an enum here. it would be a bit more consistent with other pipelines (here function to apply is equivalent to `aggregation_strategy` for token classification for instance.

It does make the code a bit more verbose. We still do need to support raw strings as arguments as it's still much simpler to use than importing the enum from somewhere.

(I'm not sure it's worth it, feel free to ignore)